### PR TITLE
Fix nil-pointer panic on arm64/multi launches with missing amd64 companion image

### PR DESCRIPTION
**Pull request by hoxhaeris**
## What

Fixes a nil-pointer panic that crashes the whole bot process when launching arm64/multi clusters whose amd64 companion release image is missing, and hardens the handler goroutines so a single bad command can no longer take the service down.

## Background

arm64 and multi launches resolve **two** images: the arch-specific install payload, and an **amd64** release image used as the test-runner (`RELEASE_IMAGE_LATEST`). `ResolveImageOrVersion` looked up that amd64 companion at six sites and passed the result straight into `pullSpecForTagRef` **without a nil check**, so whenever the companion could not be found it dereferenced a nil `*imagev1.TagReference` and panicked.

Two real cases were observed in production:

- **`launch 4.23.0-0.nightly-arm64-... arm64,aws`** — the amd64 `4.23.0-0.nightly` stream currently has tags but none `Accepted`, so `findNewestImageSpecTagWithStream` legitimately returns nil.
- **`launch 4.22 aws,arm64`** — the stable-semver path looked up the companion with `findNewestImageSpecTagWithStream(amd64IS, "4.22")`, which matches on a release named literally `"4.22"` (never exists — stable releases are named `<major>-stable`). It returned nil even though amd64 `4.22.x` exists.

Because the panic happened in bare goroutines spawned by the event and modal handlers — which `net/http` cannot recover — one bad command crashed the entire process and left the bot unresponsive for **all** users while it restarted (observed: 26 restarts, no response to the user).

## Changes

- **`pkg/manager/manager.go`**
  - Guard every amd64 `runTag` lookup so a missing companion returns a clear, user-facing error (e.g. *"no accepted amd64 4.23.0-0.nightly release is available to use as the arm64 test runner image"*) instead of panicking.
  - Resolve the stable-semver companion with `findNewestStableImageSpecTagBySemanticMajor(amd64IS, unresolved, "amd64")`, matching how the arch-specific tag is resolved — so `launch 4.22 ...,arm64` now actually resolves and launches.
  - Make `pullSpecForTagRef` nil-safe as a backstop.
- **`cmd/ci-chat-bot/slack.go`** — add a `recover()` to the event-handling goroutine.
- **`pkg/slack/modals/common/recover.go`** — new `SafeGo` helper (recovers + logs stack), applied to all modal interaction goroutines so a bug in one command can no longer crash the whole bot.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved resilience of background Slack operations by recovering from unexpected failures without terminating the bot.
  - Added diagnostic logging for recovered errors.
  - Improved image and version resolution with safer handling of missing references and clearer errors when compatible images are unavailable.
  - Preserved existing modal workflows while making asynchronous authentication, launch, listing, and cluster-management actions more reliable.

### CODE REVIEW

**Review by thiagoalessio — APPROVED** [review]
/lgtm

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -1248,6 +1248,9 @@ func buildPullSpec(namespace, tagName, isName string) string {
 }
 
 func pullSpecForTagRef(tag *imagev1.TagReference, namespace, isName string) string {
+	if tag == nil {
+		return ""
+	}
```
[the remark this answers is not in the document]
Comment (hoxhaeris) [reply pkg/manager/manager.go:1253]: Thanks — both findings were valid pre-existing bugs and are addressed in f3d0dcd0:

- **QCI runner reference**: the runSpec path now falls back to `DockerImageReference` when the amd64 status tag has an empty `Image`, matching the install-side handling, so it no longer builds a trailing-colon `registry.ci.openshift.org/ocp/release:` reference.
- **`findSpecTagByName` companion**: for arm64/multi it now returns the missing-companion error instead of reusing the arch-specific `installSpec` as the runner image.

The `recover.go#L14-23` anchor required no change — it was where the "run `make verify`" note was attached. `make verify lint test all` was run (lint is currently unavailable locally due to a golangci-lint/Go toolchain version mismatch, but `go vet -tags gcs` passes along with the `-race` tests and build).

**Comment by hoxhaeris** [comment]
Note:

- The **crashes** (`launch 4.22 aws,arm64` and a `4.23` arm64 nightly) come from a long-standing corner case in the amd64 companion-image resolution for arm64/multi launches — it's been in `ResolveImageOrVersion` since 2022 but only reachable when a bare `major.minor` arm64 launch falls through the nightly/ci branches to the stable-semver branch. Recent release-stream timing (4.22 becoming stable, 4.23 arm64 nightlies landing before their amd64 counterparts are accepted) finally routed real launches into that path. Since the launch runs in a bare goroutine, the resulting `nil` deref took the whole bot down — hence the `recover()`/`SafeGo` hardening here too.
- The **QCI `DockerImageReference` fallback** on the runner side just extends the QCI-aware handling added in b7033b5 to the mirrored amd64 runner lookup — the install side already does exactly this, so this brings the two halves back in sync.

**Review by bradmwilliams — APPROVED** [review]
